File: flight-service/src/main/java/com/rbu/service/FlightService.java
package com.rbu.service;

import com.rbu.entity.Flight;
import java.util.List;

public interface FlightService {
	  Flight save(Flight flight);

	    Flight findByCode(int code);

	    List<Flight> findByCarrier(String carrier);

	    List<Flight> findByRoute(String source, String destination);

	    List<Flight> findByPriceRange(int min, int max);

	    List<Flight> list();

	    void delete(int code);

}

File: flight-frontend/src/services/FlightService.js
import axios from "axios";

const BASE_URL = "http://localhost:8089/flights";

// GET ALL FLIGHTS
export const getAllFlights = () =>
    axios.get(`${BASE_URL}/list`);


// SAVE FLIGHT
export const saveFlight = (flight) =>
    axios.post(`${BASE_URL}/save`, flight);


// DELETE FLIGHT
export const deleteFlight = (code) =>
    axios.delete(`${BASE_URL}/${code}`);


// SEARCH BY CODE
export const getFlightByCode = (code) =>
    axios.get(`${BASE_URL}/${code}`);


// SEARCH BY CARRIER
export const getFlightByCarrier = (carrier) =>
    axios.get(`${BASE_URL}/carrier/${carrier}`);


// SEARCH BY PRICE RANGE
export const getFlightsByPriceRange = (min, max) =>
    axios.get(`${BASE_URL}/price?min=${min}&max=${max}`);

export const getFlightsByRoute = (source, destination) =>
    axios.get(
        `${BASE_URL}/route?source=${source}&destination=${destination}`
    );